model/project_parser_db.go

```diff
@@ -61,56 +65,101 @@ func ParserProjectUpdateOne(query interface{}, update interface{}) error {
 	)
 }
 
-// UpsertWithConfigNumber inserts project if it DNE, otherwise updates if the config number is correct
-func (pp *ParserProject) UpsertWithConfigNumber(num int) error {
+func setAllFieldsUpdate(pp *ParserProject) interface{} {
+	return bson.M{
+		"$set": bson.M{
+			ParserProjectEnabledKey:           pp.Enabled,
+			ParserProjectStepbackKey:          pp.Stepback,
+			ParserProjectPreErrorFailsTaskKey: pp.PreErrorFailsTask,
+			ParserProjectBatchTimeKey:         pp.BatchTime,
+			ParserProjectOwnerKey:             pp.Owner,
+			ParserProjectRepoKey:              pp.Repo,
+			ParserProjectRepoKindKey:          pp.RepoKind,
+			ParserProjectRemotePathKey:        pp.RemotePath,
+			ParserProjectBranchKey:            pp.Branch,
+			ParserProjectIdentifierKey:        pp.Identifier,
+			ParserProjectDisplayNameKey:       pp.DisplayName,
+			ParserProjectCommandTypeKey:       pp.CommandType,
+			ParserProjectIgnoreKey:            pp.Ignore,
+			ParserProjectPreKey:               pp.Pre,
+			ParserProjectPostKey:              pp.Post,
+			ParserProjectTimeoutKey:           pp.Timeout,
+			ParserProjectCallbackTimeoutKey:   pp.CallbackTimeout,
+			ParserProjectModulesKey:           pp.Modules,
+			ParserProjectBuildVariantsKey:     pp.BuildVariants,
+			ParserProjectFunctionsKey:         pp.Functions,
+			ParserProjectTaskGroupsKey:        pp.TaskGroups,
+			ParserProjectTasksKey:             pp.Tasks,
+			ParserProjectExecTimeoutSecsKey:   pp.ExecTimeoutSecs,
+			ParserProjectLoggersKey:           pp.Loggers,
+			//ParserProjectAxesKey:              pp.Axes,
+			ParserProjectConfigNumberKey: pp.ConfigUpdateNumber,
+		},
+	}
+}
+
+// TryInsert suppresses the error of inserting if it's a duplicate key error and attempts to
+// upsert if config number matches
+func (pp *ParserProject) TryUpsert() error {
+	err := pp.Insert()
+	if err == nil || !strings.Contains(err.Error(), "duplicate key error") {
+		return errors.Wrapf(err, "database error inserting parser project")
+	}
+
+	q := bson.M{
+		ParserProjectConfigNumberKey: bson.M{"$exists": false},
```

Comment:
These lines say the number both doesn't exist and is some value.

Reply:
sorry this is all wrong, will fix and test.